packages/components/src/components/hds/popover-primitive/index.js

```diff
@@ -0,0 +1,245 @@
+          popoverId = guidFor(this);
+          this.popoverElement.id = popoverId;
+        }
+        this.toggleElement.setAttribute('popovertarget', popoverId);
+      }
+
+      // this should be an extremely edge case, but in the case the popover needs to be initially forced to be open
+      // we need to use the "manual" state to support the case of multiple "menus" opened at the same time
+      // IMPORTANT! if a "popover" is set to "open" with a "manual" state, then it can't be closed via `esc` and `click outside`
+      if (this.args.isOpen) {
+        this.popoverElement.popover = 'manual';
+        this.popoverElement.showPopover();
+      } else {
+        this.popoverElement.popover = 'auto';
+      }
+
+      // Register "onBeforeToggle" + "onToggle" callback functions to be called when a native 'toggle' event is dispatched
+      registerEvent(this.popoverElement, [
+        'beforetoggle',
+        this.onBeforeTogglePopover,
+      ]);
+      registerEvent(this.popoverElement, ['toggle', this.onTogglePopover]);
+
+      // we need to spread the argument because if it's set via `{{ hash … }}` Ember complains when we overwrite one of its values
+      const anchoredPositionOptions = { ...named.anchoredPositionOptions };
+
+      // we overwrite the "strategy" if the Popover API is not supported (polyfill applied for the first time) of it's already been polyfilled (see above)
+      // this is specifically done for Firefox: currently it doesn't support it, but will soon (we need Firefox 127 to support the last 2 versions)
+      // see: https://wiki.mozilla.org/Release_Management/Release_owners
+      if (!isPopoverApiSupported() || isPopoverApiPolyfilled()) {
+        // when using the "absolute" strategy, the presence of a parent with "relative" position leads to wrong layout rendering (known issue in the polyfill library)
+        // see: https://github.com/oddbird/popover-polyfill/tree/main?tab=readme-ov-file#caveats
+        anchoredPositionOptions.strategy = 'fixed';
+      }
+
+      // Apply the `hds-anchored-position` modifier to the "popover" element
+      // (notice: this function runs the first time when the element the modifier was applied to is inserted into the DOM, and it autotracks while running.
+      // Any tracked values that it accesses will be tracked, including the arguments it receives, and if any of them changes, the function will run again)
+      // This modifiers uses the Floating UI library to provide:
+      // - positioning of the "popover" in relation to the "toggle"
+      // - collision detection (optional)
+      next(() => {
+        anchoredPositionModifier(
+          this.popoverElement, // element the modifier is attached to
+          [this.toggleElement], // positional arguments
+          anchoredPositionOptions // named arguments
+        );
+      });
+    },
+    { eager: true }
+  );
+
+  @action
+  showPopover() {
+    this.popoverElement.showPopover();
+  }
```

Comment:
I suggest we do some error handling on the `showPopover`, `hidePopover`, `togglePopover` web API methods (a try/catch) – in case the 'bubble' element doesn't exist (e.g. gets removed for some reasons) it will likely throw a hard error.

Reply:
@alex-ju would you expect something like this:
```
@action
showPopover() {
  try {
    this.popoverElement.showPopover();
  } catch (error) {
    warn(
      'The invocation of `showPopover` for the popover element caused an unexpected error.',
      {
        id: 'hds-popover.show-popover-action.invocation-failed',
        error: error,
      }
    );
  }
}
```
or like this?
```
@action
showPopover() {
  if (this.popoverElement) {
    this.popoverElement.showPopover();
  }
}
```
or something else?

Reply:
OK, done. See my last commit.